src/main/java/control/TimelineController.java

```diff
@@ -87,10 +106,93 @@ public void shotChangedHandler(CameraShotBlockUpdatedEvent event) {
         // Adjust model
         shot.setBeginCount(changedBlock.getBeginCount());
         shot.setEndCount(changedBlock.getEndCount());
-        // Remove shot from previous timeline and add to new one
-        previousTimeline.removeShot(shot);
-        this.project.getCameraTimelines()
-                .get(changedBlock.getTimetableNumber()).addShot(shot);
+
+        CameraTimeline newCameraTimeline = this.project.getCameraTimelines()
+                .get(changedBlock.getTimetableNumber());
+
+        // Remove shot from previous timeline and add to new one if changed
+        if (event.getOldTimelineNumber() != changedBlock.getTimetableNumber()) {
+            previousTimeline.removeShot(shot);
+            newCameraTimeline.addShot(shot);
+        }
+
+        // check for collisions
+        checkCollisions(changedBlock.getTimetableNumber(),
+                event.getOldTimelineNumber(), changedBlock);
+    }
+
+    /**
+     * Check collisions for new camerablock.
+     * @param timelineNumber - the timelinenumber where the camerashotblock is added
+     * @param cameraShotBlock - the camerashotblock to check collisions with
+     */
+    private void checkCollisions(int timelineNumber, CameraShotBlock cameraShotBlock) {
```

Comment:
This principle is called overloading and is very useful. This way I can make 2 methods that basically do the same thing but with other parameters without duplicating code.